vivisect/qt/remote.py

```diff
@@ -119,17 +120,18 @@ def getServer(self):
         return wsserver
 
 
-@vq_main.idlethread
 def openServerAndWorkspace(vw, parent=None):
     dia = VivConnectServerDialog(vw, parent=parent)
     host = dia.getServer()
     if host is None:
         return
+    if ':' in host:
+        host, port = host.split(':')
 
     connServerAndWorkspace(vw, str(host), parent=parent)
 
-@vq_main.workthread
-def connServerAndWorkspace(vw, host,parent=None):
+
+def connServerAndWorkspace(vw, host, port=16500, parent=None):
```

Comment:
Is a default port needed here? It is already in `connectToServer` and there it is using the `viv_port` variable (which has the same value).

Reply:
Normally yes, but I've gotten bitten in the past for changing the API on a minor release, so I've been wary of directly changing APIs to force a new required parameter

Reply:
I think my comment was a bit confusing. I was thinking about `port=None` to avoid repeating the value of `viv_port`. In case that the default port is changed at some point, this one will be likely forgotten. But I guess there are no plans to change the default port, so this may not be important.

Reply:
I like the suggestion of not repeating the port. Though in this (very specific) case, I'm less worried about that, mostly because the viv_port value hasn't changed in several years, and there are no plans (that I know of) to change it.

But I can still clean it up a bit, even if it's not strictly super important right now.